File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    period = []
    
    for progress, speed in zip(progresses, speeds):
        item = 100 - progress
        if item % speed == 0:
            period.append(item // speed)
        else:
            period.append(item // speed + 1)
    
    idx = 0
    for i in range(len(period)):
        if period[idx] < period[i]:
            print(period[idx], period[i])
            answer.append(i - idx)
            idx = i
    
    answer.append(len(period) - idx)
    
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public List<Integer> solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        int maxPercent = 100;
        
        List<Integer> time = new ArrayList<>();
        for (int i = 0; i < speeds.length; i++) {
            int remaining = maxPercent - progresses[i];
            int days = (int) Math.ceil((double) remaining / speeds[i]);
            time.add(days);
        }
        
        int count = 1;
        int currentMax = time.get(0);
        for (int i = 1; i < time.size(); i++) {
            if (time.get(i) <= currentMax) {
                count ++;
            } else {
                answer.add(count);
                count = 1;
                currentMax = time.get(i);
            }
        }
        
        answer.add(count);
        return answer;
    }
}
